uktrade/data-hub-frontend
# Axios: Update add company functional test

## Description of change

Adds functional tests for adding companies as part of the work to replace requests with axios.

The existing `add-company-spec.js` was quite difficult to debug as a result of excessive context nesting - it also depended on events from previous tests, which is an anti-pattern as described in the cypress docs (https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests). I have flattened the contexts to make it easier to work out what is going on and also to make sure that each test can be run on its own (ie with `it.only`). This means that new tests do not need to depend on the long sequence of test steps given previously. This article explains some of the reasons that we should avoid context nesting in tests: https://kentcdodds.com/blog/avoid-nesting-when-youre-testing

Adds tests for:
- when DnB search fails
- successful DnB search
- countries appear in the overseas countries list
- when DnB company creation fails
- when DnB investigation post fails

## Test instructions

Functionally everything should be the same. You could try running some of the refactored tests with `.only` to make sure that they run independently of other tests.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
